Source/DedicatedServer/Private/ServerConsoleLinux.cpp

```diff
@@ -1,23 +1,244 @@
 // Copyright 2004-2016 YaS-Online, Inc. All Rights Reserved.
 
 #include "DedicatedServerPrivatePCH.h"
+#if PLATFORM_LINUX
+#include <iostream>
+#include <sstream>
+#include <string>
+#endif
 
 void DumpConsoleHelp();
 
 #if WITH_SERVER_CODE && PLATFORM_LINUX
 	void FServerConsole::Tick()
 	{
+		int hInputEvents = getch();
+		if ( hInputEvents != ERR )
+		{
+			if ( hInputEvents == '\n' )
+			{
+				auto& sInput = m_sInput;
+				AsyncTask( ENamedThreads::GameThread, [this, sInput]()
+				{
+					TCHAR sOutput[MAX_SPRINTF] = TEXT("");
+					if ( sInput.StartsWith( TEXT( "help" ) ) )
+					{
+						ClearInputLine();
+						m_sInput.Empty();
+
+						DumpConsoleHelp();
+
+						attron( COLOR_PAIR( 1 ) );
+						FCString::Sprintf( sOutput, TEXT( "> %s%s" ), *sInput, LINE_TERMINATOR );
+						printf( "%s", TCHAR_TO_ANSI( sOutput ) );
+						refresh();
+						attroff( COLOR_PAIR( 1 ) );
+
+						m_hCommandHistory.Add(sInput);
+
+						RedrawInputLine();
+					}
+					else if ( GEngine->Exec( GWorld, *sInput ) )
+					{
+						ClearInputLine();
+						m_sInput.Empty();
+
+						attron( COLOR_PAIR( 1 ) );
+						FCString::Sprintf( sOutput, TEXT( "> %s%s" ), *sInput, LINE_TERMINATOR );
+						printf( "%s", TCHAR_TO_ANSI( sOutput ) );
+						refresh();
+						attroff( COLOR_PAIR( 1 ) );
+
+						m_hCommandHistory.Add( sInput );
+
+						RedrawInputLine();
+					}
+					else if ( GWorld->GetAuthGameMode() != nullptr && GWorld->GetAuthGameMode()->ProcessConsoleExec( *sInput, *GLog, nullptr ) )
+					{
+						ClearInputLine();
+						m_sInput.Empty();
+
+						attron( COLOR_PAIR( 1 ) );
+						FCString::Sprintf( sOutput, TEXT( "> %s%s" ), *sInput, LINE_TERMINATOR );
+						printf( "%s", TCHAR_TO_ANSI( sOutput ) );
+						refresh();
+						attroff( COLOR_PAIR( 1 ) );
+
+						m_hCommandHistory.Add( sInput );
+
+						RedrawInputLine();
+					}
+					else
+					{
+						ClearInputLine();
+						m_sInput.Empty();
+
+						attron( COLOR_PAIR( 1 ) );
+						FCString::Sprintf( sOutput, TEXT( "Unknown Command: %s%s" ), *sInput, LINE_TERMINATOR );
+						printf( "%s", TCHAR_TO_ANSI( sOutput ) );
+						refresh();
+						attroff( COLOR_PAIR( 1 ) );
+
+						RedrawInputLine();
+					}
+				});
+			}
+			else if ( hInputEvents == '\b' )
+			{
+				if ( m_sInput.Len() >= 1 )
+				{
+					m_sInput.RemoveAt( m_sInput.Len() - 1 );
+					m_sInput.AppendChar(' ');
+					RedrawInputLine();
+					m_sInput.RemoveAt( m_sInput.Len() - 1 );
+					SetCursorPosition( coords( LINES - 1, m_sInput.Len() ) );
+				}
+			}
+			else if ( hInputEvents == 27 ) // Escape Key Event
```

Comment:
Instead of a comment, maybe try to find a `KEY_ESCAPE` macro or so?

Reply:
There isn't a macro for that key, that's why I used the 27 code character. I put the comment to explain this.